File: Models/THA.Model.Base/IBaseModel.cs
using System;
using THA.Model.Core;

namespace THA.Model.Base
{
   public interface IBaseModel
   {
      Guid ID { get; set; }
      Guid CREATE_BY { get; set; }
      DateTime CREATE_DATE { get; set; }
      Guid EDIT_BY { get; set; }
      DateTime EDIT_DATE { get; set; }
      Statuses STATUS { get; set; }
   }
}

File: frontend/src/app/core/models/i-base.model.ts
import { Statuses } from './statuses.enum';
export interface IBaseModel {
  ID: string; //int
  STATUS: Statuses; //Statuses
  CREATEBY: number; //int
  EDITBY: number; //int
  CREATEDATE: Date;
  EDITDATE: Date;
}
